web-app/prisma/schema.prisma

```diff
@@ -353,23 +354,98 @@ enum AgentListType {
+  FAILED
+}
+
+enum CreditTransactionType {
+  TOP_UP
+  REFERRAL
+  SPEND
+  REFUND
+  MANUAL
+}
+
+model FiatPayment {
+  id                String             @id @default(cuid())
+  createdAt         DateTime           @default(now())
+  updatedAt         DateTime           @updatedAt
+  amount            BigInt // Amount in cents/smallest currency unit
+  currency          String // e.g., "usd"
+  status            FiatPaymentStatus
+  servicePaymentId  String             @unique // Stripe's payment_intent ID
+  serviceCustomerId String? // Stripe's customer ID
+  metadata          Json? // Additional payment metadata
+  creditTransaction CreditTransaction?
+  service           FiatPaymentService @default(STRIPE)
+}
+
+enum FiatPaymentService {
+  STRIPE
+}
+
+enum FiatPaymentStatus {
+  PENDING
+  SUCCEEDED
+  FAILED
+  REFUNDED
+  CANCELLED
+}
+
 model Job {
-  id                String    @id @default(cuid())
-  createdAt         DateTime  @default(now())
-  updatedAt         DateTime  @updatedAt
-  onChainIdentifier String    @unique
-  agentId           String
-  agent             Agent     @relation(fields: [agentId], references: [id])
-  userId            String
-  user              User      @relation(fields: [userId], references: [id])
-  status            JobStatus
-  input             String
-  output            String?
-  startedAt         DateTime  @default(now())
-  finishedAt        DateTime?
-  cost              BigInt
-  fee               BigInt    @default(0)
-  paymentTxId       String?
+  id                        String             @id @default(cuid())
+  createdAt                 DateTime           @default(now())
+  updatedAt                 DateTime           @updatedAt
+  agentJobId                String
+  paymentId                 String
+  onChainIdentifier         String             @unique
+  agentId                   String
+  agent                     Agent              @relation(fields: [agentId], references: [id])
+  userId                    String
+  user                      User               @relation(fields: [userId], references: [id])
+  status                    JobStatus
+  input                     String
+  output                    String?
+  startedAt                 DateTime           @default(now())
+  finishedAt                DateTime?
+  cost                      CreditTransaction  @relation("JobCost", fields: [creditTransactionId], references: [id])
+  refundCreditTransaction   CreditTransaction? @relation("JobRefund", fields: [refundCreditTransactionId], references: [id])
+  refundCreditTransactionId String?            @unique
+  paymentTxId               String?
+  creditTransactionId       String             @unique
+  errorNote                 String?
+  errorNoteKey              String?
+
+  blockchainIdentifier      String?
+  submitResultTime          DateTime?
+  unlockTime                DateTime?
+  externalDisputeUnlockTime DateTime?
+  agentIdentifier           String?
```

Comment:
What is `agentIdentifier` for? We have the connection to the agent, which has the `agentIdentifier`? 🤔

Reply:
Resolved https://github.com/masumi-network/sokosumi/pull/208/commits/c737666945b528fe98185b993b49143e90821578